crates/hir/src/source_analyzer.rs

```diff
@@ -212,8 +212,10 @@ impl SourceAnalyzer {
     ) -> Option<(Type, Option<Type>)> {
         let pat_id = self.pat_id(pat)?;
         let infer = self.infer.as_ref()?;
-        let coerced =
-            infer.pat_adjustments.get(&pat_id).and_then(|adjusts| adjusts.last().cloned());
+        let coerced = infer
+            .pat_adjustments
+            .get(&pat_id.as_pat()?)
```

Comment:
This might need to handle the expression case (and lookup in the expression adjustments etc) but I am not entirely sure. Otherwise trying to lookup the type of `FOO` in pattern `FOO..` will fail right? Either way would need to confirm whether thats relevant or not (by checking some IDE feature that makes use of this `type_of_pat` here).